skimage/color/tests/test_colorconv.py

```diff
@@ -645,20 +646,99 @@ def test_gray2rgb_alpha():
     x = np.random.random((5, 5, 4))
     with expected_warnings(['Pass-through of possibly RGB images']):
         assert_equal(gray2rgb(x, alpha=None).shape, (5, 5, 4))
-    with expected_warnings(['Pass-through of possibly RGB images']):
+    with expected_warnings(['Pass-through of possibly RGB images',
+                            'alpha argument is deprecated']):
         assert_equal(gray2rgb(x, alpha=False).shape, (5, 5, 3))
-    with expected_warnings(['Pass-through of possibly RGB images']):
+    with expected_warnings(['Pass-through of possibly RGB images',
+                            'alpha argument is deprecated']):
         assert_equal(gray2rgb(x, alpha=True).shape, (5, 5, 4))
 
     x = np.random.random((5, 5, 3))
     with expected_warnings(['Pass-through of possibly RGB images']):
         assert_equal(gray2rgb(x, alpha=None).shape, (5, 5, 3))
-    with expected_warnings(['Pass-through of possibly RGB images']):
+    with expected_warnings(['Pass-through of possibly RGB images',
+                            'alpha argument is deprecated']):
         assert_equal(gray2rgb(x, alpha=False).shape, (5, 5, 3))
-    with expected_warnings(['Pass-through of possibly RGB images']):
+    with expected_warnings(['Pass-through of possibly RGB images',
+                            'alpha argument is deprecated']):
         assert_equal(gray2rgb(x, alpha=True).shape, (5, 5, 4))
 
-    assert_equal(gray2rgb(np.array([[1, 2], [3, 4.]]),
-                          alpha=True)[0, 0, 3], 1)
-    assert_equal(gray2rgb(np.array([[1, 2], [3, 4]], dtype=np.uint8),
-                          alpha=True)[0, 0, 3], 255)
+    with expected_warnings(["alpha argument is deprecated"]):
+        assert_equal(gray2rgb(np.array([[1, 2], [3, 4.]]),
+                              alpha=True)[0, 0, 3], 1)
+    with expected_warnings(["alpha argument is deprecated"]):
+        assert_equal(gray2rgb(np.array([[1, 2], [3, 4]], dtype=np.uint8),
+                              alpha=True)[0, 0, 3], 255)
+
+
+@pytest.mark.parametrize("shape", [(5, 5), (5, 5, 4), (5, 4, 5, 4)])
+def test_gray2rgba(shape):
+    # nD case
+    img = np.random.random(shape)
+    rgba = gray2rgba(img)
+
+    # Shape check
+    assert_equal(rgba.shape, shape + (4, ))
+
+    # dtype check
+    assert rgba.dtype == img.dtype
+
+    # RGB channels check
+    for channel in range(3):
+        assert_equal(rgba[..., channel], img)
+
+    # Alpha channel check
+    assert_equal(rgba[..., 3], 1.0)
+
+
+def test_gray2rgba_dtype():
+    img_f64 = np.random.random((5, 5))
+    img_f32 = img_f64.astype('float32')
+    img_u8 = img_as_ubyte(img_f64)
+    img_int = img_u8.astype(int)
+
+    for img in [img_f64, img_f32, img_u8, img_int]:
+        assert gray2rgba(img).dtype == img.dtype
+
+
+def test_gray2rgba_alpha():
```

Comment:
There's no check for the image *content* here, only alpha, which is why it passes incorrectly.

Reply:
@stefanv, any numerical array like is valid input for `gray2rgba`, so I don't see any non obvious test for this purpose...

Reply:
```
assert_equal(output[..., :3] = gray2rgb(input))
```

Reply:
The requested test are implemented now @stefanv. Thank you for your suggestion.